packages/flutter_tools/lib/src/base/fingerprint.dart

```diff
@@ -120,8 +120,8 @@ class Fingerprint {
     final String version = content['version'];
     if (version != FlutterVersion.instance.frameworkRevision)
       throw new ArgumentError('Incompatible fingerprint version: $version');
-    _checksums = content['files'] ?? <String, String>{};
-    _properties = content['properties'] ?? <String, String>{};
+    _checksums = content['files']?.cast<String,String>() ?? <String, String>{};
```

Comment:
It is Map<String, dynamic> that needs cast to <String, String> right? Would it be better to loosen that <String, String> to <String, dynamic> here instead?

Reply:
/cc @cbracken

Reply:
We never use it for anything but String-String mappings, but JSON is String-dynamic. I think either way is fine.

Reply:
going by `_checksum` and `_properties` are declared as  `Map<String,String>`

@cbracken, is there a possibility of the entry being a map?

Reply:
@aam not sure I follow -- the JSON is only ever read/written by this class itself. It's a map of String->String by design. If it's not, there's been a programmer error.

Personally I prefer keeping the type as <String,String> and casting here as it indicates that this is an intentional file format decision and this isn't just arbitrary JSON. If that assumption ever changes, we can update the code.

That leaves the question of what happens if the file is some form of JSON that isn't purely String->String. I don't know that we should treat that any differently from any other sort of file corruption (e.g. replacing the contents with a PNG).

Reply:
I think I normally prefer to do type checking when retrieving particular value from json map.
So you read json into `Map<String, dynamic> _checksums`(that doesn't need typecasting), when you retrieve a value you do `String checksum = _checksums[key];`.

What you describe makes sense too. Perhaps worth adding as a comment next to the explicit cast.

Reply:
Sorry Alex, merged before I saw your comment.